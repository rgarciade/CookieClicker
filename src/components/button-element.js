import {LitElement,html,css} from "lit";
import {buttonStyle} from "./styles/button-style.js";

export class ButtonElement extends LitElement {
    static get styles() {
        return [buttonStyle];
    }

    render() {
        return html`
            
            <button  @click="${this._onClick}"><slot></slot></button>
        `;
    }
}

window.customElements.define('button-element', ButtonElement);
