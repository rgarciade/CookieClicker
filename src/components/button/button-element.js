import { LitElement, html } from 'lit';
import { buttonStyle } from './button-style.js';

export class ButtonElement extends LitElement {
	static get styles() {
		return [buttonStyle];
	}
	/**
	 * @property {Number} width - Width of the button.
	 */
	static get properties() {
		return {
			width: { type: String },
			type: { type: String },
			lengthType: { type: String },
		};
	}

	constructor() {
		super();
		this.buttonTypes = {
			primary: 'primary',
			secondary: 'secondary',
		};
		this.buttonLengthType = {
			long: 'long',
			short: 'short',
		};
	}

	buttonType = () => {
		const typeCss = this.buttonTypes[this.type] ?? this.buttonTypes['primary'];
		const buttonLength =
			this.buttonLengthType[this.lengthType] ?? this.buttonLengthType['long'];
		return `${typeCss} ${buttonLength}`;
	};

	render() {
		return html`
			<button
				class="${this.buttonType()}"
				?disabled="${this.disabled}"
				@click="${this._onClick}"
			>
				<slot></slot>
			</button>
		`;
	}
}

window.customElements.define('button-element', ButtonElement);
