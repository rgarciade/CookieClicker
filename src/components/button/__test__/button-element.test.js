import { html, fixture } from '@open-wc/testing';
import '../button-element.js';
import { getAllElementsAndShadowRoots } from 'shadow-dom-testing-library';
import { expect } from '@jest/globals';

describe('button-element', () => {
	it('matches the snapshot', async () => {
		const el = await fixture(html`<button-element></button-element>`);
		const data = getAllElementsAndShadowRoots(el, '*');
		expect(data).toMatchSnapshot();
	});
	it('matches the snapshot slot', async () => {
		const el = await fixture(
			html`<button-element><p>example-p</p></button-element>`
		);
		const data = getAllElementsAndShadowRoots(el, '*');
		expect(data).toMatchSnapshot();
	});
});
