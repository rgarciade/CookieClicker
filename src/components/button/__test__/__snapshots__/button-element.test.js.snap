// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button-element matches the snapshot 1`] = `
[
  <button-element />,
  ShadowRoot {},
  <button
    class="primary long"
  >
    
				
    <slot />
    
			
  </button>,
  <slot />,
  <style>
    
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}

  </style>,
]
`;

exports[`button-element matches the snapshot slot 1`] = `
[
  <button-element>
    <p>
      example-p
    </p>
  </button-element>,
  ShadowRoot {},
  <p>
    example-p
  </p>,
  <button
    class="primary long"
  >
    
				
    <slot />
    
			
  </button>,
  <slot />,
  <style>
    
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}

  </style>,
]
`;
